python/triumf_onlysim.py

```
@@ -86,6 +89,42 @@ def add_chunk_info(df):
   for i,r in cycle.iterrows():
     df.loc[df.start == i, 'max_pressure'] = r.total_flow
 
+def stats_for_repeated_cycles(adf, variable='total_flow') :
+    ''' 
+    This function assumed that the simulator DataFrame has been pre-processed
+    to include integer indexing and that the start times for each cycle have been 
+    calculated. This loops through the given DataFrame and 
+    1: Checks that there really is a one-to-one correspondence between dtc and 
+    the integer indexing
+    2: Finds the series for the variable in question for a given integer index 
+    since start of cycle
+    3: calculates some basic stats that can be used in plotting
+    4: Returns a DataFrame for plotting.
+    N.B. This function must be called once for each variable to be plotted.
+    ---
+    Chris.Jillings@snolab.ca 2020-04-19
+    '''
+    nstats = 7
+    di_series = adf['diindex']
+    length = di_series.max() - di_series.min()+1
+    stats_array = np.zeros((length, nstats), dtype='float64')
+    di_arr = np.arange(di_series.min(), di_series.max())
+    for i in di_arr:
+        this_series = adf.loc[adf.diindex==1, variable]
+        # Do some sanity checking here that diindex and dtc track perfectly
+        dtcmin = adf[adf.diindex==i]['dtc'].min()
+        dtcmax = adf[adf.diindex==i]['dtc'].max()
+        if ( (dtcmax-dtcmin)>1e-8 ) :
+            print("Something realy bad happened preparing stats for repeated breaths.")
+            print("There is not a one-to-one onto mapping of the integer indices and the time indices.")
+            print("Error!")
```

@jillings is printing an error enough, or you want to raise an exception so the user notices?

---

Valerio - For now I say enough and I will add the exception feature and push it this week.

---

Thank you Chris.  I agree with Valerio, it would make more sense to `raise RuntimeWarning("message")` here instead of three calls to `print`